test/aries-js-worker/test/didexchange/edgeagent_router.js

```diff
@@ -31,44 +97,187 @@ describe("DID-Exchange between an Edge Agent and a router", function() {
+    })
+
+    it(`Router is running on "${routerHttpUrl},${routerWsUrl}" with controller "${routerControllerApiUrl}"`, async function () {
+        await routerHealthCheck(routerHttpUrl, routerWsUrl, routerControllerApiUrl)
+    })
+
+    it("Alice Edge Agent receives an invitation from the router via the controller API", async function () {
+        const response = await axios.post(routerCreateInvitationPath)
+        invitation = response.data.invitation
+
+        validateInvitation(invitation)
+    })
+
+    it("Alice Edge Agent accepts the invitation from the router", function (done) {
+        acceptInvitation(aliceAgent, invitation, done)
+    })
+
+    it("Alice Edge Agent validates that the connection's state is 'completed'", function (done) {
+        aliceAgent.startNotifier(notice => {
+            try {
+                assert.isObject(notice)
+                assert.property(notice, "isErr")
+                assert.isFalse(notice.isErr)
+                assert.property(notice, "payload")
+                assert.property(notice.payload, "state")
+            } catch (err) {
+                done(err)
+            }
+            if (notice.payload.state === completedState) {
+                connectionID = notice.payload.connection_id
+
+                done()
+            }
+        }, ["connections"])
+    })
+
+    it("Alice Edge Agent sets previous connection as the router", function (done) {
+        routeRegister(aliceAgent, connectionID, done)
+    })
+
+    it("Alice Edge Agent validates that the router connection id", function (done) {
+        validateRouterConnection(aliceAgent, connectionID, done)
+    })
+
+    it("Bob Edge Agent receives an invitation from the router via the controller API", async function () {
+        const response = await axios.post(routerCreateInvitationPath)
+        invitation = response.data.invitation
+
+        validateInvitation(invitation)
+    })
+
+    it("Bob Edge Agent accepts the invitation from the router", function (done) {
+        acceptInvitation(bobAgent, invitation, done)
+    })
+
+    it("Bob Edge Agent validates that the connection's state is 'completed'", function (done) {
+        bobAgent.startNotifier(notice => {
+            try {
+                assert.isObject(notice)
+                assert.property(notice, "isErr")
+                assert.isFalse(notice.isErr)
+                assert.property(notice, "payload")
+                assert.property(notice.payload, "state")
+            } catch (err) {
+                done(err)
+            }
+            if (notice.payload.state === completedState) {
+                connectionID = notice.payload.connection_id
+
+                done()
+            }
+        }, ["connections"])
+    })
+
+    it("Bob Edge Agent sets previous connection as the router", function (done) {
+        routeRegister(bobAgent, connectionID, done)
+    })
+
+    it("Bob Edge Agent validates that the router connection id", function (done) {
```

Comment:
this description seems incomplete

Reply:
[Done](https://github.com/hyperledger/aries-framework-go/compare/5e233a431cda218ef4f3b53699aa1bf5abcd89b9..931f7cf10fee7b0a584d845740c60aa1972c7649)